sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java

```diff
@@ -169,13 +171,15 @@ public RntbdTransportClient(
             clientTelemetry,
             this.serverErrorInjector);
 
-        this.proactiveOpenConnectionsProcessor = new ProactiveOpenConnectionsProcessor(this.endpointProvider);
-        this.proactiveOpenConnectionsProcessor.init();
-
         this.id = instanceCount.incrementAndGet();
         this.tag = RntbdTransportClient.tag(this.id);
         this.channelAcquisitionContextEnabled = options.channelAcquisitionContextEnabled;
         this.globalEndpointManager = globalEndpointManager;
+        this.addressSelector = new AddressSelector(addressResolver, Protocol.TCP);
+
+        this.proactiveOpenConnectionsProcessor = new ProactiveOpenConnectionsProcessor(this.endpointProvider, this.addressSelector);
+        this.proactiveOpenConnectionsProcessor.init();
```

Comment:
Not related to this PR: but I am wondering if `init()` fails for some reason, we will currently fail the RntbdTransportClient creation. I understand that it might be the correct behavior, but we should verify that we fail gracefully.

Reply:
A failure here will bubble up to `RxDocumentClientImpl` which essentially will trigger `close()` on the following instances - `StoreClientFactory` -> `RntbdTransportClient` -> `ProactiveOpenConnectionProcessor`

Reply:
Tracking this through an issue - https://github.com/Azure/azure-sdk-for-java/issues/35905